smt/applications/tests/test_ego.py

```diff
@@ -464,6 +464,45 @@ def test_find_best_point(self):
         x, _ = ego._find_best_point(xdoe, ydoe, enable_tunneling=False)
         self.assertAlmostEqual(6.5, float(x), delta=1)
 
+    def test_ego_gek(self):
+        from smt.problems import TensorProduct
+
+        class TensorProductIndirect(TensorProduct):
+            def _evaluate(self, x, kx):
+                assert kx is None
+                _self = super(TensorProductIndirect, self)
+                response = _self._evaluate(x, kx)
+                sens = np.hstack(_self._evaluate(x, ki) for ki in range(x.shape[1]))
+                return np.hstack((response, sens))
+
+        fun = TensorProductIndirect(ndim=2, func="tanh")
+
+        # Construction of the DOE
+        sampling = LHS(xlimits=fun.xlimits, criterion="m")
```

Comment:
Maybe you should set `random_state` parameter when creating LHS instance as well to get result reproducibility.
```python
sampling = LHS(xlimits=fun.xlimits, criterion="m", random_state=0)
```
then fix the test assertion if needed